### PR TITLE
Swaps

### PR DESCRIPTION
**Pull request by 86667**

### CODE REVIEW

File: client-wasm/src/lib.rs [hunk 1]
```diff
@@ -7,8 +7,16 @@ extern crate floating_duration;
 extern crate serde_derive;
 extern crate serde;
 extern crate serde_json;
+extern crate rand;
+extern crate uuid;
+extern crate schemars;
+extern crate rocket_okapi;
```
Comment (86667) [inline client-wasm/src/lib.rs:13]: I dont think we need all these

File: client-wasm/src/swap_js_wrapper.rs [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+    /// Generate new BSTSenderData for Swap
+    pub fn setup() -> Result<JsValue, JsValue> {
+        let p: GE = ECPoint::generator(); // gen
+        // let x = FE::new_random(); // priv
+        let x: FE = ECScalar::from(&BigInt::from(1));
+
+        let q = p * x; //pub
+
+        // let k: FE = FE::new_random();
+        let k: FE = ECScalar::from(&BigInt::from(2));
```
Comment (86667) [inline client-wasm/src/swap_js_wrapper.rs:136]: still using set u,v,x,k

File: client-wasm/src/swap_js_wrapper.rs [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+        )
+    }
+}
+
+#[wasm_bindgen]
+pub struct SwapTokenW;
+
+/// Struct defines a Swap. This is signed by each participant as agreement to take part in the swap.
+#[derive(Serialize, Deserialize, JsonSchema, Debug, Clone)]
+pub struct SwapToken {
```
Comment (86667) [inline client-wasm/src/swap_js_wrapper.rs:295]: Is this still needed? It can all be in JS i think.

File: client-wasm/src/swap_js_wrapper.rs [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+    }
+
+    pub fn sign(swap_token_str: String, proof_key_priv_str: String) -> Result<JsValue, JsValue> {
+        let key = match SecretKey::from_str(&proof_key_priv_str){
+            Ok(r)=>r,
+            Err(err) => return Err(format!("SwapTokenW - parse secret key:{}, {}",&proof_key_priv_str, err.to_string()).into()),
+        };
+
+
+        //let key: SecretKey =  match serde_json::from_str(&proof_key_priv_str){
```
Comment (86667) [inline client-wasm/src/swap_js_wrapper.rs:371]: rm if no longer needed

File: client-wasm/src/swap_js_wrapper.rs [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+        Ok(
+            json!(verify).to_string().into()
+        )
+    }
+}
+
+/*
+#[wasm_bindgen_test]
+fn test_blind_sign_structs() {
+    let msg = "Message".to_string();
```
Comment (86667) [inline client-wasm/src/swap_js_wrapper.rs:426]: rm

**Review by 86667 — COMMENTED** [review]
Okay stopped reviewing half way through, there is tonnes of print statements, commented out code, unused code etc. This all needs to be cleaned up before merge

File: client-wasm/src/lib.rs (hunk 1 above)
Comment (lawlawlaw) [reply client-wasm/src/lib.rs]: Ok, removed

File: client-wasm/src/swap_js_wrapper.rs (hunk 2 above)
Comment (lawlawlaw) [reply client-wasm/src/swap_js_wrapper.rs]: Ok, now random

File: client-wasm/src/swap_js_wrapper.rs (hunk 3 above)
Comment (lawlawlaw) [reply client-wasm/src/swap_js_wrapper.rs:280]: it's more convenient to use rust here, I think.

File: client-wasm/src/swap_js_wrapper.rs (hunk 4 above)
Comment (lawlawlaw) [reply client-wasm/src/swap_js_wrapper.rs]: Ok

File: client-wasm/src/swap_js_wrapper.rs (hunk 5 above)
Comment (lawlawlaw) [reply client-wasm/src/swap_js_wrapper.rs]: Ok

File: client-wasm/src/swap_js_wrapper.rs (hunk 3 above)
Comment (86667) [reply client-wasm/src/swap_js_wrapper.rs:280]: Why, where is this used?

Comment (lawlawlaw) [reply client-wasm/src/swap_js_wrapper.rs:280]: In SwapTokenW

Comment (86667) [reply client-wasm/src/swap_js_wrapper.rs:280]: Okay, and where is SwapTokenW used?

Comment (lawlawlaw) [reply client-wasm/src/swap_js_wrapper.rs:280]: I see. It's not used. Deleting

File: client-wasm/Cargo.toml [hunk 6]
```diff
@@ -29,6 +30,11 @@ web-sys = { version = "0.3.45", features = [
   'RequestDestination',
   'Window']}
 console_error_panic_hook = "0.1.6"
+hex = "0.4.2"
+rand = { version = "0.6.5", features = ["wasm-bindgen"] }
+schemars = { version = "0.8.0-alpha-4", features = ["chrono", "uuid"] }
```
Comment (86667) [inline client-wasm/Cargo.toml:35]: Are these needed? schemars, uuid, rocket_okapi

File: src/settings.json [hunk 7]
```diff
@@ -1,4 +1,7 @@
 {
-  "min_anon_set": 10000,
-  "network": "testnet"
+  "min_anon_set": 3,
+  "state_entity_endpoint": "http://0.0.0.0:8000",
```
Comment (86667) [inline src/settings.json:3]: rm custom config settings

**Review by 86667 — COMMENTED** [review]
Code is full of console.logs and debug logs. I dont think we should include these in released version

**Review by tomt1664 — APPROVED** [review]

**Pull request merged**